test/integration/vaultdynamicsecret_integration_test.go

```diff
@@ -50,12 +45,9 @@ func TestVaultDynamicSecret(t *testing.T) {
 	tempDir, err := os.MkdirTemp(os.TempDir(), t.Name())
 	require.Nil(t, err)
 
-	tfDir, err := files.CopyTerraformFolderToDest(
-		path.Join(testRoot, "vaultdynamicsecret/terraform"),
-		tempDir,
-		"terraform",
-	)
-	require.Nil(t, err)
+	tfDir := copyTerraformDir(t, path.Join(testRoot, "vaultdynamicsecret/terraform"), tempDir)
+	copyModulesDir(t, tfDir)
+	chartDestDir := copyChartDir(t, tfDir)
```

Comment:
I'm curious why this is working off a copy of the chart dir, does the test modify the chart in some way?

Reply:
By working from a copy, it allows you to be able to modiy the chart while a test is running without affecting that test run. The test should never modify the contents of the chart.